### PR TITLE
Utility code and examples in preparation for #4

### PR DESCRIPTION
**Pull request by azaretsky**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@ FOURMOLU_EXTENSIONS := -o -XTypeApplications -o -XTemplateHaskell -o -XImportQua
 
 NIX_SOURCES := $(shell find * -not -path 'dist-newstyle/*' -iname '*.nix')
 CABAL_SOURCES := $(shell find * -not -path 'dist-newstyle/*' -iname '*.cabal')
-HASKELL_SOURCES := $(shell find * -not -path 'dist-newstyle/*' -iname '*.hs')
+HASKELL_SOURCES := $(shell find app src test -iname '*.hs')
```
Comment (nazrhom) [inline Makefile:5]: Why is this changed?

Comment (azaretsky) [reply Makefile:5]: It turned out to be an artifact of initially having some scratch `.hs` files in my working copy, instead of proper code and tests. I'll revert it back.

File: test/Tx/DummyContract.hs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import PlutusTx.Prelude
+
+type Gesture = ()
+type Redeemer = ()
+
+{-# INLINEABLE mkDummyValidator #-}
+mkDummyValidator :: Gesture -> Redeemer -> Ledger.ScriptContext -> Bool
+mkDummyValidator () () _ = True
+
+data RPS
```
[the remark this answers is not in the document]
Comment (azaretsky) [reply test/Tx/DummyContract.hs:22]: Oh, good catch!

File: test/Tx/Spec.hs [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+        (serialiseToCBORHex <$> lockTxBodyBuilder) @?= Right lockTxBodyCBOR
+    , testCase "spend a script output" $ do
+        pparamsResult <- eitherDecodeFileStrict' "devnet/protocol-parameters.json"
+        let txBodyResult = pparamsResult >>= spendTxBodyBuilder
+        (serialiseToCBORHex <$> txBodyResult) @?= Right spendTxBodyCBOR
+    , testCase "sign a transaction" $
+        (serialiseToCBORHex . signTx skey <$> lockTxBodyBuilder) @?= Right lockSignedTxCBOR
+        {-
+          -- a more realistic example with the key in a file:
+          testCaseInfo "sign a transaction" $ do
```
[the remark this answers is not in the document]
Comment (azaretsky) [reply test/Tx/Spec.hs:33]: Solely because I haven't learnt yet what is the proper way to use data files in tests: where to put them in the project tree, what paths to use to reference them, etc.

[the remark this answers is not in the document]
Comment (azaretsky) [reply test/Tx/Spec.hs:33]: It would probably be better also to externalize all the tx blobs.
